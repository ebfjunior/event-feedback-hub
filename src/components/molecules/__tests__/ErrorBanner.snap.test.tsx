import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { ErrorBanner } from '../ErrorBanner';

describe('ErrorBanner (snapshot)', () => {
	it('renders and triggers retry', () => {
		const onRetry = vi.fn();
		const { container, getByText } = render(<ErrorBanner message="Failed" onRetry={onRetry} />);
		expect(container.firstChild).toMatchSnapshot();
		fireEvent.click(getByText('Retry'));
		expect(onRetry).toHaveBeenCalled();
	});
});


