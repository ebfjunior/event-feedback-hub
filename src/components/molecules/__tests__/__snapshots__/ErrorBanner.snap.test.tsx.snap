// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ErrorBanner (snapshot) > renders and triggers retry 1`] = `
<div
  class="flex items-center justify-between rounded-md border border-destructive/40 bg-destructive/5 p-3 text-sm"
>
  <div
    class="text-destructive"
  >
    Failed
  </div>
  <button
    class="inline-flex items-center justify-center whitespace-nowrap font-medium transition-colors focus-visible:outline-none focus-visible:ring-1 disabled:pointer-events-none disabled:opacity-50 bg-destructive text-destructive-foreground shadow-sm hover:bg-destructive/90 h-8 rounded-md px-3 text-xs"
  >
    Retry
  </button>
</div>
`;
